aptos-move/framework/src/natives/cryptography/algebra/constants.rs

```diff
@@ -0,0 +1,140 @@
+        ),
+        Some(Structure::BLS12381G1Affine) => ark_constant_op_internal!(
+            context,
+            ark_bls12_381::G1Projective,
+            zero,
+            gas_params.ark_bls12_381_g1_proj_infinity * NumArgs::one()
+        ),
+        Some(Structure::BLS12381G2Affine) => ark_constant_op_internal!(
+            context,
+            ark_bls12_381::G2Projective,
+            zero,
+            gas_params.ark_bls12_381_g2_proj_infinity * NumArgs::one()
+        ),
+        Some(Structure::BLS12381Gt) => ark_constant_op_internal!(
+            context,
+            ark_bls12_381::Fq12,
+            one,
+            gas_params.ark_bls12_381_fq12_one * NumArgs::one()
+        ),
+        _ => Err(SafeNativeError::Abort {
+            abort_code: MOVE_ABORT_CODE_NOT_IMPLEMENTED,
+        }),
+    }
+}
+
+pub fn one_internal(
+    gas_params: &GasParameters,
+    context: &mut SafeNativeContext,
+    ty_args: Vec<Type>,
+    mut _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    let structure_opt = structure_from_ty_arg!(context, &ty_args[0]);
+    abort_unless_arithmetics_enabled_for_structure!(context, structure_opt);
+    match structure_opt {
+        Some(Structure::BLS12381Fr) => ark_constant_op_internal!(
+            context,
+            ark_bls12_381::Fr,
+            one,
+            gas_params.ark_bls12_381_fr_one * NumArgs::one()
+        ),
+        Some(Structure::BLS12381Fq12) => ark_constant_op_internal!(
+            context,
+            ark_bls12_381::Fq12,
+            one,
+            gas_params.ark_bls12_381_fq12_one * NumArgs::one()
+        ),
+        Some(Structure::BLS12381G1Affine) => ark_constant_op_internal!(
+            context,
+            ark_bls12_381::G1Projective,
+            generator,
+            gas_params.ark_bls12_381_g1_proj_generator * NumArgs::one()
+        ),
+        Some(Structure::BLS12381G2Affine) => ark_constant_op_internal!(
+            context,
+            ark_bls12_381::G2Projective,
+            generator,
+            gas_params.ark_bls12_381_g2_proj_generator * NumArgs::one()
+        ),
+        Some(Structure::BLS12381Gt) => {
+            context.charge(gas_params.ark_bls12_381_fq12_clone * NumArgs::one())?;
+            let element = *Lazy::force(&BLS12381_GT_GENERATOR);
+            let handle = store_element!(context, element);
+            Ok(smallvec![Value::u64(handle as u64)])
+        },
+        _ => Err(SafeNativeError::Abort {
+            abort_code: MOVE_ABORT_CODE_NOT_IMPLEMENTED,
+        }),
+    }
+}
+
+pub fn order_internal(
+    _gas_params: &GasParameters,
+    context: &mut SafeNativeContext,
+    ty_args: Vec<Type>,
+    mut _args: VecDeque<Value>,
+) -> SafeNativeResult<SmallVec<[Value; 1]>> {
+    assert_eq!(1, ty_args.len());
+    let structure_opt = structure_from_ty_arg!(context, &ty_args[0]);
+    abort_unless_arithmetics_enabled_for_structure!(context, structure_opt);
+    match structure_opt {
```

Comment:
What about the scalar field? Shouldn't we match that here too to support it?

Reply:
Fixed.